distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestPlatform.java

```diff
@@ -0,0 +1,187 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.test;
+
+import java.io.File;
+import java.util.Dictionary;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.felix.cm.file.ConfigurationHandler;
+import org.codice.ddf.platform.util.ConfigurationPropertiesComparator;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+import org.osgi.service.cm.Configuration;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+
+import ddf.common.test.BeforeExam;
+import ddf.common.test.config.ConfigurationPredicate;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class TestPlatform extends AbstractIntegrationTest {
+
+    private static final XLogger LOGGER = new XLogger(LoggerFactory.getLogger(TestPlatform.class));
+
+    private static final String testConfig = "/itest.platform.config";
+
+    private static final String testConfig2 = "/itest.platform.modified.config";
+
+    private static final long POLLER_WAIT_TIME = TimeUnit.SECONDS.toMillis(20);
+
+    private static final ConfigurationPropertiesComparator CONFIGURATION_PROPERTIES_COMPARATOR = new ConfigurationPropertiesComparator();
+
+    private static Dictionary<String, Object> configProperties;
+
+    private static Dictionary<String, Object> modifiedConfigProperties;
+
+    @BeforeExam
+    @SuppressWarnings("unchecked")
+    public void beforeExam() throws Exception {
+        configProperties = ConfigurationHandler.read(getClass().getResourceAsStream(testConfig));
+        modifiedConfigProperties = ConfigurationHandler
+                .read(getClass().getResourceAsStream(testConfig2));
+        getAdminConfig().setLogLevels();
+        getServiceManager().waitForAllBundles();
+    }
+
+    @Test
+    public void testStartUpWithExistingConfigFile() throws Exception {
+        // TODO
+    }
+
+    @Test
+    public void testModifyConfigFileFromAdmin() throws Exception {
+
+    }
+
+    @Test
+    public void testDeleteConfigFileFromAdmin() throws Exception {
+
+    }
+
+    @Test
+    public void testCreateUntrackedConfigFileFromAdmin() throws Exception {
+
+    }
+
+    @Test
+    public void testRecreationOfTrackedFile() throws Exception {
+        deleteConfigFileFromFileSystem();
+
+        getAdminConfig().waitForConfiguration("itest.platform", new ConfigurationDeleted(),
+                POLLER_WAIT_TIME);
+        LOGGER.debug("Configuration deleted!");
+        /*The next line (the one updating properties) will cause the test to fail.
+          For some reason it appears to generate a CM_LOCATION_CHANGED event, which
```

Comment:
Getting the configuration of a PID will rebind the location to the bundle asking for the configuration.  When the pax exam probe asks for the configuration, it will change location to the probe's location.  The listener should not fail for any of the possible events including CM_LOCATION_CHANGED.